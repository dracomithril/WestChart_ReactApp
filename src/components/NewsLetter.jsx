/**
 * Created by Gryzli on 21.10.2017.
 */
import React from 'react';
import PropTypes from "prop-types";
import {Image, Tooltip, OverlayTrigger} from "react-bootstrap";
import utils from "../utils";

export default class NewsLetter extends React.Component {
    componentWillUnmount() {
        console.log('component NewsLetter unmounted');
    }

    componentDidMount() {
        console.log('component NewsLetter did mount');
    }

    render() {
        const newsLetter = this.props.data;
        const today = new Date();
        const today_week= utils.weekInfo(today);
        const show = newsLetter.map((elem) => {
            let create_date = new Date(elem.created_time);
            const {weekNumber}= utils.weekInfo(create_date);
            return (<div style={{padding: 2, display: "block", border: "1px black solid"}} key={elem.id}>
                <input type={"checkbox"}/>
                <span hidden>{elem.id}</span>
                <OverlayTrigger placement={"bottom"} overlay={<Tooltip id={"tt_" + elem.id}>{elem.from_user}</Tooltip>}>
                    <Image src={utils.getFbPictureUrl(elem.from.id)}/>
                </OverlayTrigger>

                <div>
                    <span>{create_date.toLocaleDateString()}</span><br/>
                    <span>week:{weekNumber}</span><br/>
                    <span>is added</span>
                </div>
            </div>)
        });
        return (<div>
            <h4>We have {today_week.weekNumber} week of {today.getFullYear()}</h4>
            <div style={{display:"inline-flex"}}>
            {show}
            </div>
        </div>);
    }
}
NewsLetter.propTypes = {
    data: PropTypes.array
};
NewsLetter.contextTypes = {
    store: PropTypes.object
};