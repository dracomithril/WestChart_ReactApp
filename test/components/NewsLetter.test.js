/**
 * Created by XKTR67 on 10/21/2017.
 */
import React from 'react';
import PropTypes from 'prop-types';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';
import NewsLetter from "../../src/components/WestLetter";
const initial_state =require('./../data/initial_state.json');
const westletters_state =require('./../data/WestLetter.json');
const mockStore = configureMockStore([]);
describe('<WestLetter/>', () => {
    it('renders without crashing ChartPresenter', () => {
        const store = mockStore(initial_state);
        const wrapper = shallow(
            <NewsLetter data={[]}/>, {
                context: {store},
                childContextTypes: {store: PropTypes.object}
            }
        );
        expect(shallowToJson(wrapper)).toMatchSnapshot();

    });
    it('renders users', () => {
        const store = mockStore(initial_state);
        const wrapper = shallow(
            <NewsLetter data={westletters_state}/>, {
                context: {store},
                childContextTypes: {store: PropTypes.object}
            }
        );
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

});