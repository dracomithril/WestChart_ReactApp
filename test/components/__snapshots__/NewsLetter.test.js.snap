// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewsLetter/> renders users 1`] = `
<div>
  <div>
    <span>
      1707149242852457_1938133309754048
    </span>
    <img
      src="https://external.xx.fbcdn.net/safe_image.php?d=AQA4vhd4mqW12H4V&w=720&h=720&url=https%3A%2F%2Fi.ytimg.com%2Fvi%2FKL1XR6kT_40%2Fhqdefault.jpg&cfs=1&_nc_hash=AQCzRB4ci2NToCWh"
    />
    <span />
  </div>
  <div>
    <span>
      1707149242852457_1941491366084909
    </span>
    <img
      src="https://scontent.xx.fbcdn.net/v/t1.0-9/17022295_10203108927922207_1006771455463995321_n.jpg?oh=03e07f893abcd6b19475b974b05d4463&oe=59370A6C"
    />
    <span />
  </div>
</div>
`;

exports[`<NewsLetter/> renders without crashing ChartPresenter 1`] = `<div />`;
